### PR TITLE
Feature/erssup 80338

### PR DESCRIPTION
**Pull request by francisco-videira-nhs**
## Summary
* :sparkles: New Feature

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: sandbox/src/routes/r4/retrieveBinaryHelper.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    method: 'GET',
+    path: '/ObjectStore/RetrieveBinary/{fileDownloadUuid}',
+    handler: (request, h) => {
+
+      const uuid = request.params.fileDownloadUuid
+      const filename = 'example_attachment.pdf'
+      const filePath = `./r4/requestUrlForFileDownload/${filename}`
+      const responseCode = 200
+
+      if (uuid === 'd497bbe3-f88b-45f1-b3d4-9c563e4c0f5f') {
```
Comment (nhsd-jack-wainwright) [inline sandbox/src/routes/r4/retrieveBinaryHelper.js:16]: As highlighted above I wonder if it's worth supporting any ID in the legacy or UUID format here?

File: specification/components/r4/schemas/responses/retrieveBinary/307Response.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  x-correlation-id:
+    $ref: '../../headers/response/CorrelationID.yaml'
+  x-request-id:
+    $ref: '../../headers/response/RequestID.yaml'
+  Location:
+    description: The attachment location in the object store.
+    required: true
+    schema:
+      type: string
+      example: 'TODO Do we want to provide an example?'
```
Comment (nhsd-jack-wainwright) [inline specification/components/r4/schemas/responses/retrieveBinary/307Response.yaml:12]: I wonder if it's simpler to not include an example here just so clients to make any assumptions about the URLs format?

**Review by nhsd-jack-wainwright — COMMENTED** [review]
Looks good to me, just a few minor comments 🙂. I think it's probably worth including a sandbox test for both the new `retrieveBinary` and `retrieveBinaryHelper` endpoints also.

**Comment by francisco-videira-nhs** [comment]
Closing as the branch has changed.

**Pull request closed without merging**